Sources/CryptoSwift/Salsa20.swift

```diff
@@ -0,0 +1,249 @@
+//
+//  Salsa20.swift
+//  CryptoSwift
+//
+//  Created by Dennis Michaelis on 27.03.16.
+//  Copyright © 2016 Dennis Michaelis. All rights reserved.
```

Comment:
can't use it like this

Reply:
Sorry, but this is incorrect. You can not change the copyright owner, you can only commit your work to the license of the upstream project. The copyright owner will always and forever be the author of a piece of work.

See https://opensource.stackexchange.com/a/5855

Reply:
I'm referring to https://github.com/krzyzanowskim/CryptoSwift/blob/master/CONTRIBUTING.md

Reply:
Yep, this is what made me search for how others dealt with it in PRs in the first place.

I was considering contributing code, but then stumbled upon your contributing.md, which demands to make false claims about the copyright. Imo, this is inacceptable as there was the risk of you suing the original author for using their own IP in other work.

[At least in the EU](http://www.iprhelpdesk.eu/node/1841) copyright **can not be transferred** to a different person. The copyright owner will stay the original author, regardless of false copyright statements. He*she can only license their work to the project they're commiting to.

I can't say what it is like internationally, but as stated in the aforementioned stackexchange link, it is at least common practice in open source projects to handle it the way I described it.

Reply:
What I can't and don't want to accept is `All rights reserved.`. The header from the CONTRIBUTING.md has a full license along with the licensing intent in the first line. As long as the code has a history, the author's information is accessible line per line.

From a legal perspective, leaving “//  Copyright (C) Marcin Krzyżanowski <marcin@krzyzanowskim.com>” in the headers does not affect the contributors’ rights. Contributors are also listed in https://github.com/krzyzanowskim/CryptoSwift/blob/master/CONTRIBUTORS.txt

Reply:
Continuing discussion in #811, as this gets off-topic.